src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -31,45 +31,46 @@ void CableView::paint (Graphics& g)
         }
     }
 
-    g.setColour (cableColour.brighter (0.1f));
-    for (auto* cable : cables)
+    if (cableMouse != nullptr)
     {
-        auto* startEditor = board->findEditorForProcessor (cable->startProc);
-        jassert (startEditor != nullptr);
-
-        auto startPortLocation = CableViewPortLocationHelper::getPortLocation ({ startEditor, cable->startIdx, false });
-
-        if (cable->endProc != nullptr)
+        auto mousePos = cableMouse->getPosition();
+        const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->startProc);
+        if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
         {
-            auto* endEditor = board->findEditorForProcessor (cable->endProc);
-            jassert (endEditor != nullptr);
+            auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
+            drawCablePortGlow (g, endPortLocation, scaleFactor);
+        }
+    }
+}
 
-            auto endPortLocation = CableViewPortLocationHelper::getPortLocation ({ endEditor, cable->endIdx, true });
+void CableView::resized()
+{
+    for (auto* cable : cables)
+    {
+        cable->setBounds (getLocalBounds());
+    }
+}
 
-            auto startColour = startEditor->getColour();
-            auto endColour = endEditor->getColour();
-            auto levelDB = cable->endProc->getInputLevelDB (cable->endIdx);
-            drawCable (g, startPortLocation.toFloat(), startColour, endPortLocation.toFloat(), endColour, scaleFactor, levelDB);
-        }
-        else if (connectionHelper->cableMouse != nullptr)
+void CableView::mouseDown (const MouseEvent& e)
+{
+    if (e.eventComponent == nullptr)
+        return; // not a valid mouse event
```

Comment:
Interesting... was this something that was happening in your testing/debugging?

Reply:
This is leftover from the original line checking for various non intended mouse clicks. I haven't used the MouseEvent much so let me know if I misunderstood how that original logic worked.

![Screen Shot 2022-08-10 at 3 36 16 PM](https://user-images.githubusercontent.com/43660096/184024736-5b4e8ea0-c080-43fb-994e-d4e109704981.png)
.

Reply:
Actually reworked this to the original in the latest changes.

Reply:
Gotcha... I can't think why the `eventComponent` would be null, but I guess it's better to be safe than sorry.